internal/handlers/handlers.go

```diff
@@ -1,39 +1,115 @@
 package handlers
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
+	"github.com/axelx/go-yandex-metrics/internal/logger"
+	"github.com/axelx/go-yandex-metrics/internal/mgzip"
+	"github.com/axelx/go-yandex-metrics/internal/models"
+	"github.com/axelx/go-yandex-metrics/internal/mtemplate"
+	"github.com/axelx/go-yandex-metrics/internal/pg"
+	"github.com/axelx/go-yandex-metrics/internal/pg/db"
 	"github.com/axelx/go-yandex-metrics/internal/service"
 	"github.com/go-chi/chi/v5"
-	"html/template"
+	"go.uber.org/zap"
+	"io"
 	"net/http"
+	"strconv"
+	"strings"
 )
 
 type keeper interface {
 	SetGauge(string, float64) error
 	SetCounter(string, int64) error
 	GetMetric(string, string) (string, error)
+
+	SetJSONGauge(string, *float64) error
+	SetJSONCounter(string, *int64) error
+	GetJSONMetric(string, string) (models.Metrics, error)
+
 	GetTypeMetric(string) interface{}
 }
 
 type handler struct {
 	memStorage keeper
+	Logger     *zap.Logger
+	ClientDB   *db.DB
+	DBPostgres *pg.PgStorage
 }
 
-func New(k keeper) handler {
+func New(k keeper, logLevel string, newClient *db.DB, NewDBStorage *pg.PgStorage) handler {
```

Comment:
передавать следует проинициализированный логгер
```suggestion
func New(k keeper, log *zap.Logger, newClient *db.DB, NewDBStorage *pg.PgStorage) handler {
```